### PR TITLE
DFR-1995: Sending recipient in additional data - bulk print

### PR DESCRIPTION
**Pull request by rishikrsharma**
### JIRA link (if applicable) ###
DFR-1995

### Change description ###
 Sending recipient in additional data - bulk print

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManualPaymentSubmittedHandler.java [hunk 1]
```diff
@@ -50,7 +51,7 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
             log.info("Sending letter correspondence to applicant for case: {}", caseDetails.getId());
             CaseDocument caseDocument =
                 service.generateManualPaymentLetter(caseDetails, userAuthorisation, APPLICANT);
-            printService.sendDocumentForPrint(caseDocument, caseDetails);
+            printService.sendDocumentForPrint(caseDocument, caseDetails, CCDConfigConstant.APPLICANT);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManualPaymentSubmittedHandler.java:54]: Can be use the static import of APPLICANT

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 2]
```diff
@@ -168,4 +233,8 @@ private BulkPrintDocument generateRespondentCoverSheet(FinremCaseDetails caseDet
 
         return documentHelper.getCaseDocumentAsBulkPrintDocument(respondentCoverSheet);
     }
+
+    public String getRecipient(String text) {
+        return StringUtils.remove(WordUtils.capitalizeFully(text, '_'), "_");
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:238]: Can be use StringUtils.capitalize, would recon to use Springs StringUtils so no need to import apache common

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 3]
```diff
@@ -29,91 +33,138 @@ public class BulkPrintService {
...
-    @Deprecated
-    public UUID sendDocumentForPrint(final CaseDocument document, CaseDetails caseDetails) {
+    /**
+     * Please upgrade your code.
+     * This method will be removed in future versions.
+
+     * @return letterId to be return
+     * @deprecated deprecated since 15-Feb-2023
+     */
+    @Deprecated(since = "15-Feb-2023")
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:49]: We should remove this deprecated if not needed

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java [hunk 4]
```diff
@@ -36,15 +37,15 @@ public void printAssignToJudgeNotification(CaseDetails caseDetails, String authT
                 caseDetails, authToken, APPLICANT);
 
             // Send notification letter to Bulk Print
-            bulkPrintService.sendDocumentForPrint(assignedToJudgeNotificationLetter, caseDetails);
+            bulkPrintService.sendDocumentForPrint(assignedToJudgeNotificationLetter, caseDetails, CCDConfigConstant.APPLICANT);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java:40]: same here static import

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java [hunk 5]
```diff
@@ -36,15 +37,15 @@ public void printAssignToJudgeNotification(CaseDetails caseDetails, String authT
...
             log.info("Applicant notification letter sent to Bulk Print: {} for Case ID: {}", assignedToJudgeNotificationLetter,
                 caseDetails.getId());
         }
 
         if (shouldPrintNotificationForRespondentSolicitor(caseDetails)) {
             UUID respondentLetterId = bulkPrintService.sendDocumentForPrint(
                 assignedToJudgeDocumentService.generateAssignedToJudgeNotificationLetter(caseDetails, authToken, RESPONDENT),
-                caseDetails);
+                caseDetails, CCDConfigConstant.RESPONDENT);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java:48]: same here static import

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java [hunk 6]
```diff
@@ -68,12 +69,12 @@ private boolean shouldPrintNotificationForRespondentSolicitor(CaseDetails caseDe
     public void printApplicantRejectionGeneralApplication(CaseDetails caseDetails, String authToken) {
         CaseDocument applicantGeneralApplicationRejectDoc = rejectGeneralApplicationDocumentService.generateGeneralApplicationRejectionLetter(
             caseDetails, authToken, APPLICANT);
-        bulkPrintService.sendDocumentForPrint(applicantGeneralApplicationRejectDoc, caseDetails);
+        bulkPrintService.sendDocumentForPrint(applicantGeneralApplicationRejectDoc, caseDetails, CCDConfigConstant.APPLICANT);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java:72]: same here static import

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java [hunk 7]
```diff
@@ -68,12 +69,12 @@ private boolean shouldPrintNotificationForRespondentSolicitor(CaseDetails caseDe
...
     }
 
     public void printRespondentRejectionGeneralApplication(CaseDetails caseDetails, String authToken) {
         CaseDocument applicantGeneralApplicationRejectDoc = rejectGeneralApplicationDocumentService.generateGeneralApplicationRejectionLetter(
             caseDetails, authToken, RESPONDENT);
-        bulkPrintService.sendDocumentForPrint(applicantGeneralApplicationRejectDoc, caseDetails);
+        bulkPrintService.sendDocumentForPrint(applicantGeneralApplicationRejectDoc, caseDetails, CCDConfigConstant.RESPONDENT);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java:78]: same here static import

**Review by srochani — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 8]
```diff
@@ -29,91 +33,138 @@ public class BulkPrintService {
...
-    @Deprecated
-    private UUID printDocumentsWithCoversheet(CaseDetails caseDetails, BulkPrintDocument coverSheet, List<BulkPrintDocument> caseDocuments) {
+    /**
+     * Please upgrade your code.
+     * This method will be removed in future versions.
+
+     * @return sendLetterResponse response to be return
+     * @deprecated deprecated since 15-Feb-2023
+     */
+    @Deprecated(since = "15-Feb-2023")
```
[the remark this answers is not in the document]
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:141]: its in use, but marked as depreicarted

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java (hunk 6 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PaperNotificationService.java:72]: can't do- as one already with same in static import

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManualPaymentSubmittedHandler.java (hunk 1 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ManualPaymentSubmittedHandler.java:54]: can't do- as one already with same in static import

**Pull request merged**